src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/ShardedHttpExecutionServiceCluster.scala

```diff
@@ -0,0 +1,84 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import org.broadinstitute.dsde.rawls.RawlsException
+import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkflowRecord
+import org.broadinstitute.dsde.rawls.model._
+
+import scala.concurrent.Future
+import scala.util.Try
+
+class ShardedHttpExecutionServiceCluster (members: Map[Int,ExecutionServiceDAO]) extends ExecutionServiceCluster {
+
+  // ====================
+  // facade methods
+  // ====================
+
+  // only used in tests??? TODO: update tests to use the same thing runtime does, or remove this.
+  def submitWorkflow(wdl: String, inputs: String, options: Option[String], userInfo: UserInfo): Future[ExecutionServiceStatus] =
+    members.values.head.submitWorkflow(wdl, inputs, options, userInfo)
+
+  // by nature, this is only called for workflows that have not yet been submitted.
+  // therefore, we want to send the workflows to the cromwell instance chosen
+  // by the routing algorithm defined in this class.
+  def submitWorkflows[T](wdl: String, inputs: Seq[String], options: Option[String], userInfo: UserInfo)(wf: T): Future[Seq[Either[ExecutionServiceStatus, ExecutionServiceFailure]]] =
+    nextAvailableMember(wf).submitWorkflows(wdl, inputs, options, userInfo)
+
+  // following are called on a workflow that has already been submitted.
+  // therefore, we want to use the cromwell instance that has been persisted
+  // onto that workflow.
+  def status[T](id: String, userInfo: UserInfo)(wf: T): Future[ExecutionServiceStatus] =
```

Comment:
the sequence of events is going to look like this:
* call submitWorkflows, pass a batch of workflows as an argument
* route those workflows to the best cromwell instance
* persist, on each of those workflows, the cromwell instance to which it was submitted
* when calling status, pass in the Workflow/WorkflowRecord
* status() reads the Workflow/WorkflowRecords, extracts the cromwell instance that was persisted
* cluster.status() calls instance.status() on the right instance